**Notes(what needs to be done):**
1. 
2. describe available data
3.when a student has multiple claims they add a row
etc

# 00 - Inventory

**Purpose:** Summary of the raw dataset. What columns exist, what
values they take, what is missing, what one row represents. No interpretation
and no cleaning decisions here, those belong in 01_Data_Checks.

**Source:** `data/raw/CASComp Main DataSet.csv`, provided by the competition
organizers.

**Author:** Dalton | **Last run:** 2026-09-14

## 1. Head

Show the first few rows of data

In [2]:
from pathlib import Path
import pandas as pd
import numpy as np

ROOT = Path.cwd().parent
RAW = ROOT / "data" / "raw"

df = pd.read_csv(RAW / "CASComp Main DataSet.csv")
df.head()

,student_id,name,class,study,gpa,greek,off_campus,distance_to_campus,gender,sprinklered,coverage,claim_id,amount,risk_tier,training_holdout
0,1,"Saiz, Leslie",Junior,Humanities,1.203913,Non-greek,On campus,0.0,Male,True,Additional Living Expense,0,0.0,2,training
1,1,"Saiz, Leslie",Junior,Humanities,1.203913,Non-greek,On campus,0.0,Male,True,Guest Medical,0,0.0,2,training
2,1,"Saiz, Leslie",Junior,Humanities,1.203913,Non-greek,On campus,0.0,Male,True,Liability,0,0.0,2,training
3,1,"Saiz, Leslie",Junior,Humanities,1.203913,Non-greek,On campus,0.0,Male,True,Personal Property,0,0.0,2,training
4,2,"Reddy, Dawna",Junior,Science,0.609616,Non-greek,On campus,0.0,Female,True,Additional Living Expense,0,0.0,2,training


**Finding:** One row per student per coverage. Student 1
appears four times, once each for Additional Living Expense, Guest Medical,
Liability and Personal Property.

## 2. Shape
How many rows and columns, and does that match one row per
student per coverage?

In [5]:
df.shape

(40071, 15)

**Findings:** 40,071 rows, 15 columns, 10,000 students. Four coverages x 10,000
students would be 40,000, so there are 71 extra rows

## 3. What are the extra rows from?


In [3]:
df.groupby('student_id').size().value_counts()

4    9933
5      63
6       4
Name: count, dtype: int64

In [4]:
sizes = df.groupby('student_id').size()
multi_ids = sizes[sizes > 4].index

multi = df[df.student_id.isin(multi_ids)].sort_values(['student_id', 'coverage'])
multi.head()

,student_id,name,class,study,gpa,greek,off_campus,distance_to_campus,gender,sprinklered,coverage,claim_id,amount,risk_tier,training_holdout
874,219,"al-Tabet, Qais",Freshman,Other,1.510015,Greek,On campus,0.0,Female,True,Additional Living Expense,0,0.000000,1,training
875,219,"al-Tabet, Qais",Freshman,Other,1.510015,Greek,On campus,0.0,Female,True,Guest Medical,0,0.000000,1,training
876,219,"al-Tabet, Qais",Freshman,Other,1.510015,Greek,On campus,0.0,Female,True,Liability,0,0.000000,1,training
872,219,"al-Tabet, Qais",Freshman,Other,1.510015,Greek,On campus,0.0,Female,True,Personal Property,40,2294.001208,1,training
873,219,"al-Tabet, Qais",Freshman,Other,1.510015,Greek,On campus,0.0,Female,True,Personal Property,41,505.570355,1,training


**Findings:** The extra rows are additional claims. This also confirms that the claims are not aggregated into 1 amount

Check for Null Values(move to data checks)

In [5]:
df.isna().sum()

student_id            0
name                  0
class                 0
study                 0
gpa                   0
greek                 0
off_campus            0
distance_to_campus    0
gender                0
sprinklered           0
coverage              0
claim_id              0
amount                0
risk_tier             0
training_holdout      0
dtype: int64

No null values